### ISSUE TITLE
build.zig.zon file .hash does not include executable bit in practice

### ISSUE THREAD

**Issue by slimsag** [ziglang/zig#16272]
### Zig Version

0.11.0-dev.3890+43c98dc11

### Steps to Reproduce and Observed Behavior

I was implementing a tool in Go to calculate/update build.zig.zon file hashes. While doing this, I noticed that the Zig implementation in `src/Package.zig` currently doesn't include executable file bit information in practice.

If I download/extract [this tarball](https://github.com/hexops/freetype/archive/26f9ea0ce8ec934d9d53caec80f6b3d0f857892b.tar.gz) using my system tar extractor, you can see we have at least two executable files in it:

```
freetype-26f9ea0ce8ec934d9d53caec80f6b3d0f857892b % ls -lah *.sh
-rwxrwxr-x  1 slimsag  staff   248B Jun 29 01:13 update.sh
-rwxrwxr-x  1 slimsag  staff   318B Jun 29 01:13 verify.sh
```

However, if I modify this code in the stdlib to print the file hash and result of `isExecutable`:

https://github.com/ziglang/zig/blob/0a6cd257b9c8a9093b966e3851dc8261e19b531a/src/Package.zig#L718

Then we find `isExecutable` always returns `false`:

```
update.sh: 3ade16b05234cffff8aa60857d4ee2f1a90698da63e908148aefe45242ae8075 - false
verify.sh: 7359f08ae3202633064963b58058ef193d2f358c28abe08c035550e3bad8571e - false
```

This appears to be due to this TODO during tar extraction:

https://github.com/ziglang/zig/blob/0a6cd257b9c8a9093b966e3851dc8261e19b531a/src/Package.zig#L603-L611

Importantly, this applies to all unix platforms too - not just windows - so once this is fixed, everyone's `build.zig.zon` file hashes will change.

### Expected Behavior

build.zig.zon hashes include executable bit information in them

**Comment by squeek502** [comment]
I believe 1\. in the TODO comment would be addressed by #15382

**Comment by andrewrk** [comment]
Note the plan for this is https://github.com/ziglang/zig/issues/17463#issuecomment-1984798880 and the hashes will be unchanged, continuing to not include the executable bit. All ELF files and files with shebang lines will be given +x regardless of the executable bit in the tarball.

**Comment by andrewrk** [comment]
Fixed by 34bb670bb695969ce57cb3fc0e1bb673a1cc7243.
